### DISCUSSION TITLE
What's the best way to install Tailwind into Angular project?

### DISCUSSION THREAD

**Discussion by sasafister** [tailwindlabs/tailwindcss#6390] [Help] [answered]
I'm struggling to install Tailwind 3.0 in Angular 13. I had no issue installing Tailwind in VueJS or React, but Angular has different setup I guess.

Anyone has some tips to share to make this smooth as possible? I would appreciate any advice.

**Comment by red2678** [comment]
I really wish we could get some sort of framework guide from either the Angular team or Tailwind team.

**Comment by therawk** [comment, marked as answer]
https://blog.angular.io/the-new-state-of-css-in-angular-bec011715ee6

Look at the Tailwind and other PostCSS section 👍

**Reply by sasafister** [reply]
@therawk I tried that but it's not working that way. There is different configuration in TW 3.0, and I did that as well, but it's not just simple like that...

**Reply by therawk** [reply]
@sasafister I actually wrote a migration guide:
 https://link.medium.com/NIxl4CbUWlb

This takes the new configuration settings into account.
What kind of problems do you run into?

**Reply by sasafister** [reply]
I just doesn't work. I tried to follow official instalaction without postcss, and add it late like you did but it wont work. This is my tailwind config

```
module.exports = {
  content: ["./app/src/**/*.{html,js,ts}"],
  theme: {
    extend: {},
  },
  plugins: [],
};
```

And here is my styles.css

```
@tailwind base;
@tailwind components;
@tailwind utilities;

html,
body {
  height: 100%;
}
body {
  margin: 0;
  font-family: Roboto, "Helvetica Neue", sans-serif;
}
```

I run ```npx tailwindcss -i ./src/styles.css -o ./dist/output.css --watch```, but JIT is not working at all. I'm little bit confused because I'm working with TW from v0.7 and never had problems with Vue, but with Angular it's really frustrating to set it up.

**Reply by therawk** [reply]
Oh now here's something. I have just created a new project with Angular v13 and tailwind 3. It seems like JIT only works the first time you start the app and then doesn't OR when you change the `tailwind.config.json` file while you're running `ng serve`.

This made me think what might be causing this and I realized, Angular now caches some data. Could you check, if you have the (hidden) .angular directory and delete it? After that, just run `ng serve` - **don't** use the command u stated (`npx tailwindcss -i ./src/styles.css -o ./dist/output.css --watch`).

If this works for you, update your `angular.json` file with the following:

```
"cli": {
    "cache": {
      "enabled": false
    }
  },
```
I added it right below the 2nd line. Below
```
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
```

Now I can even tell if JIT works or not by looking at the terminal. If this comes up (the Tailwind CSS is watching information is shown twice, it'll work. Otherwise only cached classes will work):

```
⠸ Generating browser application bundles (phase: building)...
info - Tailwind CSS is watching for changes...
info - https://tailwindcss.com/docs/just-in-time-mode#watch-mode-and-one-off-builds
⠴ Generating browser application bundles (phase: building)...
info - Tailwind CSS is watching for changes...
info - https://tailwindcss.com/docs/just-in-time-mode#watch-mode-and-one-off-builds
✔ Browser application bundle generation complete.

```

Hope that fixes it for you as well.
Also, here's a repo I created with all settings that worked for me
https://github.com/therawk/angular13-tailwind3

**Reply by sasafister** [reply]
@therawk thank you very much. I really appreciate it. It works like charm. I think that you should write blog post about this project and maybe put template for someone who is building from scratch. I've spend days to figure it out and I'm lost without Tailwind so you've come to the rescue :)

**Reply by therawk** [reply]
@sasafister hey, you're welcome! I'm happy it's working for you now 👌
I have been thinking about starting a blog recently and yeah, this might become my very first post 🥳😅

**Reply by sasafister** [reply]
I encourage you to do it. It's a great way to get some exposure and help others like me :)

**Reply by arnab8820** [reply]
@therawk I have tried your solution, but I still have the problem of new classes not being generated.

Steps I have followed:
1. delete .angular folder
2. add the following to angular.json
`"cli": {
    "cache": {
      "enabled": false
    }
  },`
3. copied over tailwind.config.js and postcss.config.js from the repo you have mentioned
4. run ng serve

One thing I have noticed is I am not seeing the following lines in terminal:
`info - Tailwind CSS is watching for changes...`
`info - https://tailwindcss.com/docs/just-in-time-mode#watch-mode-and-one-off-builds`

So what am I missing here?

**Reply by therawk** [reply]
Hey, sorry for the late reply @arnab8820.

Which version of Angular are u using?
This issue was probably specific to version 13.1.0 or earlier.  Disabling cache shouldn't be necessary anymore and following this https://tailwindcss.com/docs/guides/angular guide show actually work.

Hope this fixes your issue 🤞
